src/Transition.js

```diff
@@ -316,26 +309,20 @@ class Transition extends React.Component {
 
   onTransitionEnd(timeout, handler) {
     this.setNextCallback(handler)
+
     const node = this.props.nodeRef
       ? this.props.nodeRef.current
       : ReactDOM.findDOMNode(this)
 
-    const doesNotHaveTimeoutOrListener =
-      timeout == null && !this.props.addEndListener
-    if (!node || doesNotHaveTimeoutOrListener) {
-      setTimeout(this.nextCallback, 0)
-      return
-    }
+    if (node && this.props.addEndListener && timeout == null) {
```

Comment:
@taion why `node` is needed here?

Reply:
i think this is a legacy thing from the conversion to use `nodeRef`; if there's no associated node, then previously it was impossible to call `addEndListener`, because the first arg was expected to be the `node`.

Reply:
Is there any case when `node` may be missing?

Reply:
might be possible if e.g. the transition is rendering a child that's actually not there, maybe?